src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -578,29 +586,26 @@ private static void addAllConfigKeys(ConfigDef container, ConfigDef other, Set<S
   }
 
   /**
-   * Extract the biggest integer contained in the string
-   *
-   * @param string - the string to extract the maximum integer from
-   * @return int - the biggest integer found in the string, 0 if none found
+   * Finds all instances of `${[0-9]+}` in `topics.dir` and returns a list.
+   * @return list of all numbers found in the regex pattern
    */
-  private static int extractMaximumInt(String string) {
-    int num = 0;
-    int res = 0;
+  private List<Integer> getPartsToReplace() {
 
-    for (char c : string.toCharArray()) {
+    List<Integer> toReplace = new ArrayList<>();
+    Pattern pattern = Pattern.compile("\\$\\{\\d+}");
+    Matcher partsMatcher = pattern.matcher(getString(TOPICS_DIR_CONFIG));
 
-      // Convert into an integer while there are consecutive numeric digits
-      if (Character.isDigit(c)) {
-        num = num * 10 + (c - '0');
-      } else {
-        res = Math.max(res, num);
-
-        // Reset the number
-        num = 0;
+    while (partsMatcher.find()) {
+      String part = partsMatcher.group();
+      if (!part.isEmpty()) {
+        part = part.substring(2, part.length() - 1); // trim "${" and "}"
```

Comment:
You can use match groups for this. With something like "\\$\\{(\d+)\\}" as pattern, match group 1 will be the digits only.